lib/Event/ReorderEvent.php

```diff
@@ -30,7 +32,7 @@ class ReorderEvent extends AbstractDocumentEvent
      */
     public function __construct($document, $destId, $after)
     {
-        parent::__construct($document);
+        $this->document = $document;
```

Comment:
The point is that it already would be better than the current situation. My proposed solution would remove the `setNode` redundancy in the `PersistEvent` and `ReorderEvent`, and would give me the possibility to have a matching type hint in the `handleNodeFromRegistry` function of the `RegistratorSubscriber` (the other possibility would be to have two separate functions). It would also allow to easily use the `RegistratorSubscriber` with other events, which might come in the future.

And just because there is no intention to let user add or change these event (without saying this would be a bad idea, haven't thought about it too much), it still doesn't mean it is not polymorphic. As I already said this would make sense in the `handleNodeFromRegistry` function to have a `SetNodeInterface` (would have to come up with a better name though), and you've been already using it on your own with the `AbstractMappingEvent` type hint in some `handleHydrate` methods. There it would also be better to have a cleaner separation.